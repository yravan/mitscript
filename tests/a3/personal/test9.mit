f = fun() {
  if (true) {
    return "returned from f";
  } else {
    return "should not get this";
  }
  print("still in f");
};
print(f());

g = fun() {
  while (true) {
    return "returned from g";
  }
  print("still in g");
};
print(g());

h = fun() {};
print(h());

x = fun() {
  g = fun() {
    print("inside nested");
    return 5;
  };
  print(g());
  print("after nested");
};
x();

return 5;
print("should not run");