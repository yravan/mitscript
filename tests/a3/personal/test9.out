returned from f
returned from g
None
inside nested
5
after nested
